framework/include/kernels/ConservativeAdvection.h

```diff
@@ -28,12 +28,25 @@ class ConservativeAdvectionTempl : public GenericKernel<is_ad>
 protected:
   virtual GenericReal<is_ad> computeQpResidual() override;
   virtual Real computeQpJacobian() override;
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar) override;
   virtual void computeResidual() override;
   virtual void computeJacobian() override;
 
+  /// Scalar material property
+  const GenericMaterialProperty<Real, is_ad> & _scalar;
+
+  /// field variable
+  const VariableGradient * _field_variable;
+
+  /// field variable variable number
+  const unsigned int _field_variable_var;
+
   /// advection velocity
   const MooseArray<GenericRealVectorValue<is_ad>> * _velocity;
 
+  /// advective quantity present
```

Comment:
more info please. I look at the following data member `_adv_quant` and see that it's a reference and think that an `_adv_quant` should always be present